libraries/Microsoft.Bot.Builder/CloudAdapterBase.cs

```diff
@@ -116,144 +106,109 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
+            ValidateCointinuationActivity(continuationActivity);
 
-            if (string.IsNullOrWhiteSpace(audience))
-            {
-                throw new ArgumentNullException($"{nameof(audience)} cannot be null or white space.");
-            }
+            return ProcessProactiveAsync(CreateClaimsIdentity(botAppId), continuationActivity, null, callback, cancellationToken);
+        }
 
-            return ProcessProactiveAsync(claimsIdentity, reference, audience, callback, cancellationToken);
+        /// <inheritdoc/>
+        public override Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, Activity continuationActivity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            _ = claimsIdentity ?? throw new ArgumentNullException(nameof(claimsIdentity));
+            _ = callback ?? throw new ArgumentNullException(nameof(callback));
+            ValidateCointinuationActivity(continuationActivity);
+
+            return ProcessProactiveAsync(claimsIdentity, continuationActivity, null, callback, cancellationToken);
         }
 
-        // TODO: oauth prompt support
+        /// <inheritdoc/>
+        public override Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, Activity continuationActivity, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            _ = claimsIdentity ?? throw new ArgumentNullException(nameof(claimsIdentity));
+            _ = callback ?? throw new ArgumentNullException(nameof(callback));
+            ValidateCointinuationActivity(continuationActivity);
+
+            return ProcessProactiveAsync(claimsIdentity, continuationActivity, audience, callback, cancellationToken);
+        }
 
         /// <summary>
         /// The implementation for continue conversation.
         /// </summary>
         /// <param name="claimsIdentity">A <see cref="ClaimsIdentity"/> for the conversation.</param>
-        /// <param name="reference">A <see cref="ConversationReference"/> for the conversation.</param>
+        /// <param name="continuationActivity">The continuation <see cref="Activity"/> used to create the <see cref="ITurnContext" />.</param>
         /// <param name="audience">The audience for the call.</param>
         /// <param name="callback">The method to call for the resulting bot turn.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>A task that represents the work queued to execute.</returns>
-        protected async Task ProcessProactiveAsync(ClaimsIdentity claimsIdentity, ConversationReference reference, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
+        protected async Task ProcessProactiveAsync(ClaimsIdentity claimsIdentity, Activity continuationActivity, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
```

Comment:
Should this be setting CallerId?

```cs
continuationActivity.CallerId = await GenerateCallerIdAsync(claimsIdentity).ConfigureAwait(false);
```

Reply:
you're suggestion makes sense to me - let's ask @garypretty - the idea here was just to allow the continuation activity to be completely controlled by the application - the implication being that it has to get it right - but perhaps that is asking too much in this particular aspect

Reply:
I'll make this change now anyhow.

Reply:
The current code in BotFrameworkAdapter just calls GetContinuationActivity on a ConersationReference and that's all - that function doesn't set the CallerId so I think that it's never been set for this continuation activity - at least not as far as I can see.